search.go

```diff
@@ -352,6 +352,8 @@ type SearchResult struct {
 	Referrals []string
 	// Controls are the returned controls
 	Controls []Control
+	// Cookie is the returned cookie
+	Cookie []byte
```

Comment:
What is the purpose of this new field? The cookie is saved and can be obtained through the DirSync Control struct

Reply:
I thought that if I wanted to store a cookie in an external program, I cannot get a cookie in the external program because DirSync Control struct is not defined in the external program. Is there a way to get cookie with an external program?

Reply:
By that definition, what's the difference to the SearchResult struct, which isn't defined "in external programs" either? If someone wanted to extract the cookie, one could use `FindControl` and cast the result into the `*ControlDirSync` to access the cookie field from the control
```
var cookie []byte
for doMore {
	res, err := conn.DirSync(req, DirSyncObjectSecurity, 1000, cookie)
	if err != nil {
		log.Fatalf("failed to search: %s", err)
	}
	for _, entry := range res.Entries {
		entry.Print()
	}
	ctrl := FindControl(res.Controls, ControlTypeDirSync)
	if ctrl == nil || ctrl.(*ControlDirSync).Flags == 0 {
		doMore = false
	}
	cookie = ctrl.(*ControlDirSync).Cookie // <--
}
```

Reply:
I understood. My validation code was not good, I fixed it not to include it in SearchResult. Thank you very much.